job_templates/ci_job.xml.em

```diff
@@ -176,7 +176,9 @@ echo "# BEGIN SECTION: docker info"
 docker info
 echo "# END SECTION"
 echo "# BEGIN SECTION: Build Dockerfile"
-@[if os_name == 'linux-armhf']@
+@[if turtlebot_demo]@
```

Comment:
Rather than add another branch here, you can use the existing `os_name=='linux'` invocation and just specify the `INSTALL_TURTLEBOT2_DEMO_DEPS` arg with something like: 

```
@[elif os_name == 'linux']@
docker build @(--build-arg INSTALL_TURTLEBOT2_DEMO_DEPS='yes' if turtlebot_demo) -t ros2_batch_ci_turtlebot_demo linux_docker_resources
@[else]@